File: instrumentation/common-api/src/main/java/io/opentelemetry/android/instrumentation/common/ApplicationStateListener.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.instrumentation.common;

/**
 * Listener interface that is called whenever the instrumented application is brought to foreground
 * from the background, or vice versa.
 *
 * <p>This class is internal and is hence not for public use. Its APIs are unstable and can change
 * at any time.
 */
public interface ApplicationStateListener {

    /**
     * Called whenever the application is brought to the foreground (i.e. first activity starts).
     */
    void onApplicationForegrounded();

    /** Called whenever the application is brought to the background (i.e. last activity stops). */
    void onApplicationBackgrounded();
}

File: services/src/main/java/io/opentelemetry/android/internal/services/applifecycle/ApplicationStateListener.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.android.internal.services.applifecycle

/**
 * Listener interface that is called whenever the instrumented application is brought to foreground
 * from the background, or vice versa.
 *
 *
 * This class is internal and is hence not for public use. Its APIs are unstable and can change
 * at any time.
 */
interface ApplicationStateListener {
    /**
     * Called whenever the application is brought to the foreground (i.e. first activity starts).
     */
    fun onApplicationForegrounded()

    /**
     * Called whenever the application is brought to the background (i.e. last activity stops).
     */
    fun onApplicationBackgrounded()
}
